tests/deliver_to_nonexistent_contract.go

```diff
@@ -0,0 +1,165 @@
+	subnetBTeleporterMessenger, err := teleportermessenger.NewTeleporterMessenger(
+		teleporterContractAddress, subnetBInfo.ChainRPCClient,
+	)
+	Expect(err).Should(BeNil())
+
+	//
+	// Fund the deployer address on Subnet B
+	//
+	ctx := context.Background()
+
+	fundAmount := big.NewInt(0).Mul(big.NewInt(1e18), big.NewInt(10)) // 10eth
+	fundDeployerTx := utils.CreateNativeTransferTransaction(
+		ctx, subnetBInfo, fundedAddress, fundedKey, deployerAddress, fundAmount,
+	)
+	utils.SendTransactionAndWaitForAcceptance(ctx, subnetBInfo, fundDeployerTx, true)
+
+	//
+	// Deploy ExampleMessenger to Subnet A, but not to Subnet B
+	// Send a message that should fail to be executed on Subnet B
+	//
+	_, subnetAExampleMessenger := localUtils.DeployExampleCrossChainMessenger(ctx, fundedAddress, fundedKey, subnetAInfo)
+
+	// Derive the eventual address of the destination contract on Subnet B
+	nonce, err := subnetBInfo.ChainRPCClient.NonceAt(ctx, deployerAddress, nil)
+	Expect(err).Should(BeNil())
+	destinationContractAddress, err := deploymentUtils.DeriveEVMContractAddress(deployerAddress, nonce)
+	Expect(err).Should(BeNil())
+
+	//
+	// Call the example messenger contract on Subnet A
+	//
+	message := "Hello, world!"
+	optsA := utils.CreateTransactorOpts(ctx, subnetAInfo, fundedAddress, fundedKey)
+	tx, err := subnetAExampleMessenger.SendMessage(
+		optsA,
+		subnetBInfo.BlockchainID,
+		destinationContractAddress,
+		fundedAddress,
+		big.NewInt(0),
+		big.NewInt(300000),
+		message,
+	)
+	Expect(err).Should(BeNil())
+
+	// Wait for the transaction to be mined
+	receipt, err := bind.WaitMined(ctx, subnetAInfo.ChainRPCClient, tx)
+	Expect(err).Should(BeNil())
+	Expect(receipt.Status).Should(Equal(types.ReceiptStatusSuccessful))
+
+	sendEvent, err := utils.GetEventFromLogs(receipt.Logs, subnetATeleporterMessenger.ParseSendCrossChainMessage)
+	Expect(err).Should(BeNil())
+	Expect(sendEvent.DestinationChainID[:]).Should(Equal(subnetBInfo.BlockchainID[:]))
+
+	teleporterMessageID = sendEvent.Message.MessageID
+
+	//
+	// Relay the message to the destination
+	//
+
+	receipt = network.RelayMessage(ctx, receipt, subnetAInfo, subnetBInfo, false, true)
+	receiveEvent, err := utils.GetEventFromLogs(receipt.Logs, subnetATeleporterMessenger.ParseReceiveCrossChainMessage)
+	Expect(err).Should(BeNil())
+	deliveredTeleporterMessage := receiveEvent.Message
+
+	//
+	// Check that the message was successfully relayed
+	//
+	delivered, err := subnetBTeleporterMessenger.MessageReceived(
+		&bind.CallOpts{},
+		subnetAInfo.BlockchainID,
+		teleporterMessageID,
+	)
+	Expect(err).Should(BeNil())
+	Expect(delivered).Should(BeTrue())
+
+	//
+	// Check that the message was not successfully executed
+	//
+	executionFailedEvent, err := utils.GetEventFromLogs(
+		receipt.Logs,
+		subnetBTeleporterMessenger.ParseMessageExecutionFailed,
+	)
+	Expect(err).Should(BeNil())
+	Expect(executionFailedEvent.Message.MessageID).Should(Equal(deliveredTeleporterMessage.MessageID))
```

Comment:
I think we should check against `teleporterMessageID` here instead, then we don't need the extra `receiveEvent` and `deliveredTeleporterMessage` parse from above since it's already checked in `RelayMessage`

Reply:
doesn't hurt to leave i guess though, since for other network implementations liek Fuji, the `RelayMessage` doesn't necessarily check receive event. If decided to keep, can we still check against sent `teleporterMessageID`?

Reply:
We need the `deliveredTeleporterMessage` in any case so we can retry execution later. Will leave as is since it's functionally the same.